library/src/main/java/com/airbnb/android/airmapview/DefaultAirMapViewBuilder.java

```diff
@@ -97,12 +95,7 @@ private AirMapViewBuilder getWebMapViewBuilder() {
   }
 
   private static boolean checkNativeMapSupported(Context context) {
-    return isGooglePlayServicesAvailable(context) && hasWriteExternalStoragePermission(context);
-  }
-
-  private static boolean hasWriteExternalStoragePermission(Context context) {
-    return ContextCompat.checkSelfPermission(context,
-        Manifest.permission.WRITE_EXTERNAL_STORAGE) == PackageManager.PERMISSION_GRANTED;
+    return isGooglePlayServicesAvailable(context);
```

Comment:
Nope. Actually as of api 19 you can read and write from your own special
area of external storage without permission. But the permission was needed
for before 19. So as of 23 if your app isn't granted the permission it
still works fine.
On Sep 27, 2015 20:17, "Felipe Lima" <notifications@github.com> wrote:

> In
> library/src/main/java/com/airbnb/android/airmapview/DefaultAirMapViewBuilder.java
> <https://github.com/airbnb/AirMapView/pull/60#discussion_r40516134>:
>
> > @@ -97,12 +95,7 @@ private AirMapViewBuilder getWebMapViewBuilder() {
> >    }
> >
> >    private static boolean checkNativeMapSupported(Context context) {
> > -    return isGooglePlayServicesAvailable(context) && hasWriteExternalStoragePermission(context);
> > -  }
> > -
> > -  private static boolean hasWriteExternalStoragePermission(Context context) {
> > -    return ContextCompat.checkSelfPermission(context,
> > -        Manifest.permission.WRITE_EXTERNAL_STORAGE) == PackageManager.PERMISSION_GRANTED;
> > +    return isGooglePlayServicesAvailable(context);
>
> WRITE_EXTERNAL_STORAGE is not required anymore on Google Play Services
> 8.1?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/airbnb/AirMapView/pull/60/files#r40516134>.
>